From c2cc0444b636cf8d7c7a8f23b593a90421de543d Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Thu, 18 Jul 2013 17:01:37 +0700
Subject: Dalvik: Fix assert statement in createExceptionHandlingStream

BZ: 124399

createExceptionHandlingStream does not take into account that
backend can be in an error state happened before. As a result scheduler
does not emit new instructons and stream pointer remains the same.
So the check for start and end stream should take into account that
they are equal.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I0060eadc006bd460475c50580baaf0d5dcbee204
Orig-MCG-Change-Id: Id43159eba8ca23f90919a5d06fdea8ea9bac66c3
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/ExceptionHandling.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/ExceptionHandling.cpp b/vm/compiler/codegen/x86/ExceptionHandling.cpp
index dabef82..4c7ca3c 100644
--- a/vm/compiler/codegen/x86/ExceptionHandling.cpp
+++ b/vm/compiler/codegen/x86/ExceptionHandling.cpp
@@ -70,7 +70,7 @@ void ExceptionHandlingRestoreState::createExceptionHandlingStream(
     size_t lengthOfTargetLabel = strlen(targetLabel);
 
     // Developer needs to ensure that this doesn't happen
-    assert(end > beginning);
+    assert(end >= beginning);
 
     // Calculate size of exception handling instructions
     size_t sizeOfStream = end - beginning;
-- 
1.7.4.1

